src/View3d.ts

```diff
@@ -853,4 +874,52 @@ export class View3d {
   hasWebGL2(): boolean {
     return this.canvas3d.hasWebGL2;
   }
+
+  handleKeydown = (event: KeyboardEvent): void => {
+    // control-option-1 (mac) or ctrl-alt-1 (windows)
+    if (event.code === "Digit1" && event.altKey && event.ctrlKey) {
+      this.tweakpaneOpen = !this.tweakpaneOpen;
+      this.tweakpane.element.style.display = this.tweakpaneOpen ? "block" : "none";
+    }
+  };
+
+  removeEventListeners(): void {
+    window.removeEventListener("keydown", this.handleKeydown);
+  }
+
+  setupGui(container: HTMLElement): Pane {
+    if (this.tweakpane) {
+      this.canvas3d.containerdiv.removeChild(this.tweakpane.element);
+    }
+
+    const pane = new Pane({ title: "Advanced Settings", container });
+    const paneStyle: Partial<CSSStyleDeclaration> = {
+      position: "absolute",
+      top: "0",
+      right: "0",
+      display: "none",
+    };
+    Object.assign(pane.element.style, paneStyle);
+
+    // LIGHTS
+    const lights = pane.addFolder({ title: "Lights (isosurface)" });
+
+    const addFolderForLight = (light: ThreeLight, title: string): void => {
+      const folder = lights.addFolder({ title, expanded: false });
+      folder.addInput(light, "color", { color: { type: "float" } });
```

Comment:
My first thought for things that don't map 1-1 with an internal value, is to create an intermediate data structure like 
`tweakable:{ambientLight:{color:whateverMapsWellToTweakPane,...}, ...}`
And then in the tweakpane change handler do the transformation to set the value into the deep internal object.

Reply:
This function is mostly an anti-redundant-code measure, as lights actually work pretty reasonably in this respect. It could be unwound into four separate blocks of bindings for these lights, or removed from the body of `setupGui`.